### PR TITLE
Add support for postgresql 17, Debian 12

### LINKED ISSUE #2044: Chado installation problem
### Tripal Version

0229c91

### Drupal Version

11.0.9

### PostgreSQL Version

17.2

### PHP Version

8.3.14

### General Description

Error occurs during Chado 1.3 installation

### Steps to reproduce

Install  drupal, istall tripal from git.
Datastorage -> Chado -> Install Chado
Run job

### Error messages

```shell
[almalinux@licebase2-alma tripal]$ /var/www/tripal4/vendor/bin/drush trp-run-jobs --job_id=4 --username=lb_admin --root=/var/www/tripal4/web

2024-12-02 14:55:22
Tripal Job Launcher
Running as user 'lb_admin'
-------------------
2024-12-02 14:55:22: Job ID 4.
2024-12-02 14:55:22: Calling: tripal_chado_install_chado(chado, 1.3)
 [error]  ERROR: SQLSTATE[42883]: Undefined function: 7 ERROR:  function create_point(integer, bigint) does not exist
LINE 1: SELECT box (create_point(0, $1), create_point($2,500000000))
                    ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
QUERY:  SELECT box (create_point(0, $1), create_point($2,500000000))
CONTEXT:  SQL function "boxrange" during inlining
[site http://default] [TRIPAL] ERROR: SQLSTATE[42883]: Undefined function: 7 ERROR:  function create_point(integer, bigint) does not existLINE 1: SELECT box (create_point(0, $1), create_point($2,500000000))                    ^HINT:  No function matches the given name and argument types. You might need to add explicit type casts.QUERY:  SELECT box (create_point(0, $1), create_point($2,500000000))CONTEXT:  SQL function "boxrange" during inlining
 [error]  Schema installation part 3 of 41 Failed...
Installation (Step 1 of 2) problems!
 [error]  Message: Job execution failed: Failed to complete Chado installation task.
Schema installation part 3 of 41 Failed...
Installation (Step 1 of 2) problems!
```

### Screenshots

_No response_

### LINKED ISSUE #2048: Support postgres 17
### Task Description

Add postgres 17 to testing and fix any issues that may arise

### Branch Name

tv4g0-issue2048-support-postgres-17

### PR DESCRIPTION
**Pull request by dsenalik**
# New Feature

## Description
This advances the highest version of Postgresql used for testing and dockerfiles from 16 to 17
And thanks to Lacey, it advances the debian version from 11 (bullseye) to 12 (bookworm)

## Testing?
1. See if tests pass (probably not right away as there is no Drupal dockerfile for 17 yet)
2. See if you can build a docker

### CODE REVIEW

**Comment by dsenalik** [comment]
A simple patch file is not possible, as there are multiple places where errors will occur.
I made changes to the chado 1.3 sql files themselves 😟.
Some changes are from this patch file:
https://github.com/GMOD/Chado/blob/1.4/chado/migrations/V1.3.3.007__fix_search_path.sql
which references this issue:
https://github.com/GMOD/Chado/issues/65
which explains what is going on.

The later changes are just to add public to the search path when accessing box functions

With these changes I can finally build a Postgresql 17 docker locally
pg17

File: .github/workflows/SCHEDULE-buildDrupalDocker.yml [hunk 1]
```diff
@@ -93,8 +94,8 @@ jobs:
           labels: 'drupal.version.label="${{ matrix.drupal-version }}",php.version.label="${{ matrix.php-version }}", postgresql.version.label="${{ matrix.pgsql-version }}"'
       ## Build the image tagged as latest which is the highest version combo that we feel is well supported.
       - uses: mr-smithers-excellent/docker-build-push@v6
-        name: Build latest using 10.3.x-dev, PHP 8.3, PgSQL 16
-        if: ${{ matrix.drupal-version == '10.3.x-dev' && matrix.php-version == '8.3' && matrix.pgsql-version == '16' }}
+        name: Build latest using 10.3.x-dev, PHP 8.3, PgSQL 17
+        if: ${{ matrix.drupal-version == '10.3.x-dev' && matrix.php-version == '8.3' && matrix.pgsql-version == '17' }}
```
Comment (laceysanderson) [inline .github/workflows/SCHEDULE-buildDrupalDocker.yml]: Bump the Drupal version for latest in the schedules docker drupal base image as well.
```suggestion
        name: Build latest using 11.1.x-dev, PHP 8.3, PgSQL 17
        if: ${{ matrix.drupal-version == '11.1.x-dev' && matrix.php-version == '8.3' && matrix.pgsql-version == '17' }}
```

**Review by laceysanderson — COMMENTED** [review]
Code-wise everything is looking good so far. I haven't looked through the chado schema changes yet though.

**Review by laceysanderson — CHANGES_REQUESTED** [review]
So while reviewing the chado schema changes in this PR an issue popped up:

- there are changes to the parts files that are not represented in the original default_schema file. This makes it possible that we will lose these changes in the future if the parts ever need to be regenerated.
- unfortunatly the parts-specific changes are search_path focused which means we can't move these into the original default_schema file as they will be altered by the chunking script.

There are two possible solutions for this:
1. Update the chunking script to be a lot smarter about search_path. This is harder then it sounds because when the parts are applied there are additional changes to search path added.
2. Divide the original default_schema into sections first by moving the functions, views and other schema into different files. Then we only chunk and alter the schema name for the chado portion. This allows us to control the search_path better and more holistically for the functions and views.

I'm taking a stab at the second solution right now.

**Review by laceysanderson — APPROVED** [review]
Based on time constraints, I've put the more ideal solution on hold and documented it in #2072.

The fix in this PR is effective but also means we can no longer regenerate the parts. As such, I removed the original single `default_schema-1.3.sql` file that those parts were generated from to prevent us from accidentally losing these changes.

I've also tested this and looked through it in detail and feel confident in the current solution :-) Thanks for the work @dsenalik! I believe this is ready to merge and as such, I've updated the required checks to include PostgreSQL 17 instead of PostgreSQL 16.

**Pull request merged**